### PR TITLE
V9: Add new email Model for notifications

### PR DESCRIPTION
**Pull request by nikolajlauridsen**
This PR adds a new email model to be sent with email notifications.

The reason for this is to be able to both send the email address and the email display name, since this is used in Deploy.

I opted to add a new model instead of trying to change the old one, since it doesn't make much sense for that to have the display name, since it gets converted into a MimeMessage down the line.

While I was working on this I also noticed that we added the ability to fall back to a "from email" from the configuration, however the way the code currently was, the configured "from email" was never used, so I've also amended the code so that is no longer the case.

### CODE REVIEW

File: src/Umbraco.Core/Models/Email/NotificationEmailAddress.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+namespace Umbraco.Cms.Core.Models.Email
+{
+    /// <summary>
+    /// Represents an email address used for notifications. Contains both the address and its display name.
+    /// </summary>
+    public class NotificationEmailAddress
+    {
+        public string DisplayName { get; }
+
+        public string Adress { get; }
```
Comment (AndyButland) [inline src/Umbraco.Core/Models/Email/NotificationEmailAddress.cs:10]: Typo here - `Address`.

File: src/Umbraco.Infrastructure/Extensions/EmailMessageExtensions.cs [hunk 2]
```diff
@@ -58,6 +55,58 @@ public static MimeMessage ToMimeMessage(this EmailMessage mailMessage, string co
...
+                if (notificationAddress is not null)
+                {
+                    notificationAddresses.Add(notificationAddress);
+                }
+            }
+
+            return notificationAddresses;
+        }
+
         private static void AddAddresses(MimeMessage message, string[] addresses, Func<MimeMessage, InternetAddressList> addressListGetter, bool throwIfNoneValid = false)
```
Comment (AndyButland) [inline src/Umbraco.Infrastructure/Extensions/EmailMessageExtensions.cs:110]: Just to make the file easier to read top to bottom, maybe move this helper method up to below the public method that uses it.  Or if you prefer public methods at the top, make this the first private helper method in the class.

**Review by AndyButland — APPROVED** [review]

File: src/Umbraco.Core/Models/Email/NotificationEmailAddress.cs (hunk 1 above)
Comment (nikolajlauridsen) [reply src/Umbraco.Core/Models/Email/NotificationEmailAddress.cs]: Ahh good spot, for some reason I always misspell Address 😄

File: src/Umbraco.Infrastructure/Extensions/EmailMessageExtensions.cs (hunk 2 above)
Comment (nikolajlauridsen) [reply src/Umbraco.Infrastructure/Extensions/EmailMessageExtensions.cs]: Good point, I don't think it's too big an issue, so I moved the helper method up below the public method.

Comment (AndyButland) [reply src/Umbraco.Infrastructure/Extensions/EmailMessageExtensions.cs]: No, no big deal - just think it's nice when you read a class from to to bottom that methods are in an order reflecting how they are called.  Sounds good.

Comment (nikolajlauridsen) [reply src/Umbraco.Infrastructure/Extensions/EmailMessageExtensions.cs]: Oh I agree, I was referring to the mixing of public/private methods 😄

**Pull request merged**